### PR TITLE
Issue #2431 - add support for headers in RemoteTermServiceProvider

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: fhir-server/src/main/java/com/ibm/fhir/server/listener/FHIRServletContextListener.java [hunk 1]
```diff
@@ -330,6 +331,17 @@ private void configureTermServiceCapabilities(PropertyGroup fhirConfig) throws E
                                 .build());
                         }
 
+                        Object[] headersArray = remoteTermServiceProviderPropertyGroup.getArrayProperty("headers");
```
Comment (prb112) [inline fhir-server/src/main/java/com/ibm/fhir/server/listener/FHIRServletContextListener.java:334]: javadoc?

**Review by prb112 — APPROVED** [review]
LGTM

File: fhir-server/src/main/java/com/ibm/fhir/server/listener/FHIRServletContextListener.java (hunk 1 above)
Comment (JohnTimm) [reply fhir-server/src/main/java/com/ibm/fhir/server/listener/FHIRServletContextListener.java:334]: added docs

**Review by csandersdev — APPROVED** [review]
LGTM

**Pull request merged**